src/test/lib/optimizer/strategy/chunk_pruning_rule_test.cpp

```diff
@@ -154,20 +154,20 @@ TEST_F(ChunkPruningRuleTest, MultipleOutputs2) {
 
   // clang-format off
   auto common =
-    PredicateNode::make(greater_than_(a, 123),  // allows for pruning of chunk 2
+    PredicateNode::make(greater_than_(a, 123),      // Predicate allows for pruning of chunk 2
       stored_table_node);
   auto lqp =
     UnionNode::make(SetOperationMode::All,
-      PredicateNode::make(greater_than_(b, 700),    // would allow for pruning of chunk 0
-        PredicateNode::make(less_than_(b, 850),     // would allow for pruning of chunk 3
+      PredicateNode::make(greater_than_(b, 700),    // Predicate allows for pruning of chunk 0, 2
+        PredicateNode::make(less_than_(b, 850),     // Predicate allows for pruning of chunk 3
           common)),
-      PredicateNode::make(greater_than_(b, 850),  // would allow for pruning of chunk 1
+      PredicateNode::make(greater_than_(b, 850),    // Predicate allows for pruning of chunk 0, 1, 2
         common));
   // clang-format on
 
   StrategyBaseTest::apply_rule(_rule, lqp);
   auto pruned_chunk_ids = stored_table_node->pruned_chunk_ids();
-  std::vector<ChunkID> expected_chunk_ids = {ChunkID{2}};
+  std::vector<ChunkID> expected_chunk_ids = {ChunkID{0}, ChunkID{2}};
   EXPECT_EQ(pruned_chunk_ids, expected_chunk_ids);
```

Comment:
This test has been wrong. Compare table `int_float4.tbl`, which looks as follows:
```
a|b
int|float
12345|457.7
123456|700.0
12345|456.7
123456|800.0
123|458.7
12|350.7
123456|900.0
```

@j-tr and I have no idea (yet) why this test has passed previously. 😄